tests/test_skills/test_task_subprocess.py

```diff
@@ -36,19 +36,13 @@ def setup_class(cls) -> None:
         cls.set_config("agent.task_manager_mode", "multiprocess", "str")
 
     def test_task_run_in_a_subprocess(self):
-        """Test task run in a subporocess."""
+        """Test task run in a subprocess."""
         process = self.run_agent()
-        try:
-            is_running = self.is_running(process)
-            assert is_running, "AEA not running within timeout!"
-            assert not self.missing_from_output(
-                process, ["Task id is"], 10, is_terminating=False
-            )
-            assert not self.missing_from_output(
-                process, ["result is"], 10, is_terminating=False
-            )
-        finally:
-            process.terminate()
-            process.wait(10)
-            print(self.stdout[process.pid])
-            print(self.stderr[process.pid])
+        is_running = self.is_running(process)
+        assert is_running, "AEA not running within timeout!"
+        assert not self.missing_from_output(
+            process, ["Task id is"], 10, is_terminating=False
+        )
+        assert not self.missing_from_output(
+            process, ["result is"], 10, is_terminating=False
+        )
```

Comment:
fixes `tests/test_skills/test_task_subprocess.py::TestTaskTestSkill::test_task_run_in_a_subprocess`, which threw an error because of the check of subprocess termination  introduced in `teardown_class`

Reply:
What happens to subprocesses now if assert fails?

Reply:
the inner block of the try statement containing all assertions is the same. The finally block is not needed here as it does not test anything, and in fact interferes with the check in the teardown method, similar as was the case in the other test which I refer to [above](https://github.com/valory-xyz/open-aea/pull/336#issuecomment-1255442599) and are found in this commit 8d417faf1ec1a953c45dee9d45649fb78cfd392e